src/controllers/queries/index.ts

```diff
@@ -224,13 +224,79 @@ export class QueriesController extends HTMLController {
     }
 
     return this.html(
-      this.scope3CarbonConsumptionResponseTemplates.newScope3CarbonConsumptionResponseFormPage(
-        'form',
-        connection,
+      this.scope3CarbonConsumptionResponseTemplates.newScope3CarbonConsumptionResponseFormPage({
+        formStage: 'form',
+        company: connection,
         queryId,
-        query.details['quantity'],
-        query.details['productId']
-      )
+        quantity: query.details['quantity'],
+        productId: query.details['productId'],
+      })
+    )
+  }
+
+  /**
+   * @param param.queryId:UUID - query uuid identifier
+   * @param param.companyId:UUID - connection uuid identifier
+   * @param query.partialQuery:'on' - either render partial or not (checkbox state)
+   *
+   * @returns a table of connections for partial query
+   */
+  @SuccessResponse(200)
+  @Get('/{queryId}/partial/{companyId}')
+  public async scope3CO2Partial(
+    @Path() queryId: UUID,
+    @Path() companyId: UUID,
+    @Query() partialQuery?: 'on'
+  ): Promise<HTML> {
+    this.logger.debug('partial query response requested %j', { queryId, companyId })
+    const [query]: QueryRow[] = await this.db.get('query', { id: queryId })
+    if (!query) throw new NotFoundError('query not found')
+
+    const [company]: ConnectionRow[] = await this.db.get('connection', { id: companyId })
+    if (!company) throw new NotFoundError('company connection not found')
+
+    this.logger.debug('query and connection - are found %j', { company, query })
+    const connections: ConnectionRow[] = await this.db.get('connection', { status: 'verified_both' })
+
+    // due to very long names, re-assigning to a shorter variable (render)
+    const render = this.scope3CarbonConsumptionResponseTemplates.newScope3CarbonConsumptionResponseFormPage
+    this.logger.info('rendering partial query')
+
+    return this.html(
+      render({
+        ...query.details,
+        company,
+        queryId,
+        partial: partialQuery === 'on' ? true : false,
+        connections,
+        formStage: 'form',
+      })
+    )
+  }
+
+  /**
+   * @param param.connectionId:UUID
+   * @param query.partialSelect:'on' - if it's selected by checkbox, then it would return 'on'
+   *
+   * @returns - a tabe row for partial query
+   */
+  @SuccessResponse(200)
+  @Get('/partial-select/{connectionId}/')
+  public async partialSelect(@Path() connectionId: UUID, @Query() partialSelect?: 'on'): Promise<HTML> {
+    this.logger.debug('partial select %j', { connectionId })
```

Comment:
if we're just logging our a string value liek `connectionId` just log it out as a string. It'll render much better. Something like:

```ts
this.logger.debug('partial select %s', connectionId)
```

Reply:
yes!